frontend/components/common/Table.tsx

```diff
@@ -0,0 +1,362 @@
+import { ReactElement, useEffect, useMemo, useState } from 'react'
+import Select from 'react-select'
+import {
+  useFilters,
+  useGlobalFilter,
+  usePagination,
+  useSortBy,
+  useTable,
+} from 'react-table'
+import styled from 'styled-components'
+
+import {
+  ALLBIRDS_GRAY,
+  BORDER,
+  CLUBS_GREY,
+  FOCUS_GRAY,
+  WHITE,
+} from '../../constants/colors'
+import { BORDER_RADIUS, MD, mediaMaxWidth } from '../../constants/measurements'
+import { BODY_FONT } from '../../constants/styles'
+import { Icon } from '.'
+
+const styles = {
+  control: ({ background, ...base }) => {
+    return {
+      ...base,
+      border: `1px solid ${BORDER}`,
+      boxShadow: 'none',
+      width: '100%',
+    }
+  },
+  option: ({ background, ...base }, { isFocused, isSelected }) => {
+    const isEmphasized = isFocused || isSelected
+    return {
+      ...base,
+      background: isEmphasized ? FOCUS_GRAY : background,
+      color: CLUBS_GREY,
+    }
+  },
+}
+
+type FilterOption = {
+  options: string[]
+  label: string
+  func: (a, b) => boolean
+}
+
+type Row = { [key: string]: any }
+
+type tableProps = {
+  columns: Row[]
+  data: Row[]
+  searchableColumns: string[]
+  filterOptions?: FilterOption[]
+}
+
+const Styles = styled.div`
+  padding: 1rem;
+  width: 100%;
+`
+
+const SearchWrapper = styled.div`
+  overflow: visible;
+  margin: 0px;
+  ${mediaMaxWidth(MD)} {
+    width: 100%;
+    margin: 0;
+    border-bottom: 1px solid ${BORDER};
+    background: ${WHITE};
+  }
+`
+const Toolbar = styled.div`
+  margin-bottom: 50px;
+`
+const Input = styled.input`
+  border: 1px solid ${ALLBIRDS_GRAY};
+  outline: none;
+  color: ${CLUBS_GREY};
+  width: 100%;
+  font-size: 1em;
+  padding: 8px 10px;
+  background: ${WHITE};
+  border-radius: ${BORDER_RADIUS};
+  font-family: ${BODY_FONT};
+
+  &:hover,
+  &:active,
+  &:focus {
+    background: ${FOCUS_GRAY};
+  }
+`
+
+const Table = ({
+  columns,
+  data,
+  searchableColumns,
+  filterOptions,
+}: tableProps): ReactElement => {
+  const [searchQuery, setSearchQuery] = useState<string>('')
+  const [tableData, setTableData] = useState<Row[]>([])
+  const [selectedFilter, setSelectedFilter] = useState<any>({})
+  useEffect(() => {
+    const searchedData = data.filter((item) => {
+      if (!searchQuery) return true
+      if (!searchQuery || searchQuery.length < 3) {
+        return true
+      }
+      let valid = false
+      for (const searchId in searchableColumns) {
```

Comment:
This could use the [some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) method.